coderd/database/queries/provisionerjobs.sql

```diff
@@ -107,37 +107,57 @@ FROM
 				ON fj.id = pj.id -- Ensure we retrieve full details from `provisioner_jobs`.
                                  -- JOIN with pj is required for sqlc.embed(pj) to compile successfully.
 ORDER BY
-	fj.created_at;
+	fj.created_at DESC;
 
 -- name: GetProvisionerJobsByOrganizationAndStatusWithQueuePositionAndProvisioner :many
-WITH pending_jobs AS (
-    SELECT
-        id, created_at
-    FROM
-        provisioner_jobs
-    WHERE
-        started_at IS NULL
-    AND
-        canceled_at IS NULL
-    AND
-        completed_at IS NULL
-    AND
-        error IS NULL
+WITH filtered_provisioner_jobs AS (
+	-- Step 1: Filter provisioner_jobs
+	SELECT
+		id, created_at
+	FROM
+		provisioner_jobs
+	WHERE
+		COALESCE(array_length(@ids::uuid[], 1), 0) = 0 OR id = ANY(@ids::uuid[]) -- Apply filter early to reduce dataset size before expensive JOIN
 ),
-queue_position AS (
-    SELECT
-        id,
-        ROW_NUMBER() OVER (ORDER BY created_at ASC) AS queue_position
-    FROM
-        pending_jobs
+pending_jobs AS (
```

Comment:
Nit: This CTE is not ideal for performance and I don't think  it adds much value (single WHERE statement), it could just as well be embedded into `ranked_jobs` which currently doesn't have a WHERE statement. This feels like a "disconnect of logic" even if naming a part of the query can have value.

A better name for `ranked_jobs` may be `ranked_pending_jobs` thus transferring the value of the name to the other CTE.